### PR TITLE
Core 3.0 Preview 4 notes

### LINKED ISSUE #12108: [update] .NET Core 3.0 Preview 4 Notes
Update the what's new core 3.0 article to include preview 4 notes.

### PR DESCRIPTION
**Pull request by Thraka**
## Summary

Updated the what's new for core 3.0 to include preview 4 notes.

Fixes #12108

### CODE REVIEW

File: docs/core/whats-new/dotnet-core-3-0.md [hunk 1]
```diff
@@ -47,7 +48,34 @@ Even though .NET Core 3.0 supports **.NET Standard 2.1**, the default `dotnet ne
 </Project>
 ```
 
-If you're using Visual Studio, you need Visual Studio 2019 as Visual Studio 2017 doesn't support **.NET Standard 2.1** or **.NET Core 3.0**.
+If you're using Visual Studio, you need Visual Studio 2019 as Visual Studio 2017 doesn't support **.NET Standard 2.1** or **.NET Core 3.0**. It is highly recommended that you use [Visual Studio 2019 Update 1 Preview](https://visualstudio.microsoft.com/vs/preview/).
```
Comment (rpetrusha) [inline docs/core/whats-new/dotnet-core-3-0.md]:
```suggestion
If you're using Visual Studio, you need Visual Studio 2019, as Visual Studio 2017 doesn't support **.NET Standard 2.1** or **.NET Core 3.0**. We highly recommend that you use [Visual Studio 2019 Update 1 Preview](https://visualstudio.microsoft.com/vs/preview/).
```

File: docs/core/whats-new/dotnet-core-3-0.md [hunk 2]
```diff
@@ -101,9 +129,23 @@ During `dotnet build` or `dotnet publish`, an executable is created that matches
...
+[Tiered compilation](https://devblogs.microsoft.com/dotnet/tiered-compilation-preview-in-net-core-2-1/) (TC) is on by default with .NET Core 3.0. This feature enables the runtime to more adaptively use the Just-In-Time (JIT) compiler to get better performance.
+
+The main benefit of TC is to enable (re-)jitting methods with slower-but-faster to produce code or higher-quality-but-slower to produce code. This helps increase performance of an application as it goes through various stages of execution, from startup through steady-state. This contrasts with the non-TC approach, where every method is compiled a single way (the same as the high-quality tier), which is biased to steady-state over startup performance.
 
-This feature was added as an opt-in feature in [.NET Core 2.1](https://devblogs.microsoft.com/dotnet/announcing-net-core-2-1/) and then was enabled by default in [.NET Core 2.2 Preview 2](https://devblogs.microsoft.com/dotnet/announcing-net-core-2-2-preview-2/). Subsequently, it has been reverted back to opt in with the .NET Core 2.2 release.
+To enable Quick JIT (tier 0 jitted code), use this setting in your project:
```
Comment (rpetrusha) [inline docs/core/whats-new/dotnet-core-3-0.md]:
```suggestion
To enable Quick JIT (tier 0 jitted code), use this setting in your project file:
```

File: docs/core/whats-new/dotnet-core-3-0.md [hunk 3]
```diff
@@ -101,9 +129,23 @@ During `dotnet build` or `dotnet publish`, an executable is created that matches
...
+
+```xml
+<PropertyGroup>
+  <TieredCompilationQuickJit>true</TieredCompilationQuickJit>
+</PropertyGroup>
+```
+
+To disable TC completely, use this setting in your project:
```
Comment (rpetrusha) [inline docs/core/whats-new/dotnet-core-3-0.md]:
```suggestion
To disable TC completely, use this setting in your project file:
```

File: docs/core/whats-new/dotnet-core-3-0.md [hunk 4]
```diff
@@ -213,6 +255,12 @@ Here is a C# 8.0 example of reading through the [**launch.json**](https://github
 
 Windows offers a rich native API in the form of flat C APIs, COM, and WinRT. While .NET Core supports **P/Invoke**, .NET Core 3.0 adds the ability to **CoCreate COM APIs** and **Activate WinRT APIs**. For a code example, see the [Excel Demo](https://github.com/dotnet/samples/tree/master/core/extensions/ExcelDemo).
 
+## HTTP/2 support
+
+The <xref:System.Net.Http.HttpClient?displayProperty=nameWithType> type supports the HTTP/2 protocol. Support is currently disabled but can be turned on by calling `AppContext.SetSwitch("System.Net.Http.SocketsHttpHandler.Http2Support", true);` before you use <xref:System.Net.Http.HttpClient>. You can also enable HTTP/2 support by setting the `DOTNET_SYSTEM_NET_HTTP_SOCKETSHTTPHANDLER_HTTP2SUPPORT` environment variable to `true` before you run your app.
+
+If HTTP/2 is enabled, the HTTP protocol version will be negotiated via TLS/ALPN and HTTP/2 will only be used if the server selects to use it.
```
Comment (rpetrusha) [inline docs/core/whats-new/dotnet-core-3-0.md]:
```suggestion
If HTTP/2 is enabled, the HTTP protocol version will be negotiated via TLS/ALPN, and HTTP/2 will only be used if the server selects to use it.
```

File: docs/core/whats-new/dotnet-core-3-0.md [hunk 5]
```diff
@@ -278,6 +326,10 @@ Starting with Preview 3, running .NET Core 3.0 on Linux with Docker works better
 * Explicit size can be set as an absolute number or percentage of cgroup limit.
 * Minimum reserved segment size per GC heap is 16 mb. This size reduces the number of heaps that are created on machines.
 
+## Smaller Garbage Collection heap sizes
+
+While the default Garbage Collector (GC) heap size was reduced for Docker, memory usage has been improved while not running in a container. This better aligns with the generation 0 allocation budget with modern processor cache sizes.
```
Comment (rpetrusha) [inline docs/core/whats-new/dotnet-core-3-0.md]: The connection between the while clause and the main part of the sentence is unclear.

**Review by rpetrusha — APPROVED** [review]
LGTM, @Thraka. I've left a few minor suggestions and one comment. I'll open a PR to add the AppContext switch to the managed reference.

File: docs/core/whats-new/dotnet-core-3-0.md (hunk 5 above)
Comment (Thraka) [reply docs/core/whats-new/dotnet-core-3-0.md]: Ahh yes.. A bad merge between two different sentences. I'll fix that.

**Pull request merged**